aviatrix/data_source_aviatrix_all_network_domains_test.go

```diff
@@ -0,0 +1,127 @@
+package aviatrix
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/AviatrixSystems/terraform-provider-aviatrix/v2/goaviatrix"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestAccDataSourceAviatrixAllNetworkDomains_basic(t *testing.T) {
+	rName := acctest.RandString(5)
+	charset := "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+	tgwName := acctest.RandStringFromCharSet(5, charset) + acctest.RandString(5)
+	awsSideAsNumber := "64512"
+	ndName := acctest.RandStringFromCharSet(5, charset) + acctest.RandString(5)
+	resourceName := "data.aviatrix_all_network_domains.test"
+
+	skipAcc := os.Getenv("SKIP_DATA_ALL_NETWORK_DOMAINS")
+	if skipAcc == "yes" {
+		t.Skip("Skipping Data Source All Network Domains tests as SKIP_DATA_ALL_NETWORK_DOMAINS is set")
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+		},
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceAviatrixAllNetworkDomainsConfigPreBase(rName, tgwName, awsSideAsNumber, ndName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckTgwNetworkDomainExists("aviatrix_aws_tgw_network_domain.test", tgwName, ndName),
+				),
+				Destroy: false,
+			},
+			{
+				Config: testAccDataSourceAviatrixAllNetworkDomainsConfigBase(),
```

Comment:
It should be possible to use the data source in the same config as the Network Domains are created with the correct dependencies.